api4/config.go

```diff
@@ -143,13 +143,22 @@ func updateConfig(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	err = c.App.SaveConfig(cfg, true)
+	oldCfg, newCfg, err := c.App.SaveConfig(cfg, true)
```

Comment:
Since this is the only place the returned old and new config are used, wouldn't it be better to leave the SaveConfig signature as it was and fetch the old config first?  That would leave the many other calls unchanged, where the new returns are always ignored.

Reply:
I've thought about this for a while. The problem I was trying to fix with this "extended" return was that the solution you suggest would be race-prone. Config can change at any time so potentially we could end up getting diffs for something that the call itself hasn't really touched (or no diffs at all). 

I could have added a separate method instead of changing `SaveConfig` of course but I thought it could still be causing bugs down the line to keep relying on getting the config (in response to saving it) through a separate call.

Reply:
I see.